app/tests/unit/validators/permission.spec.js

```diff
@@ -0,0 +1,186 @@
+const jestJoi = require('jest-joi');
+expect.extend(jestJoi.matchers);
+
+const schema = require('../../../src/validators/permission').schema;
+const { uuidv4, uuidv4MultiModel, stringMultiModel } = require('../../../src/validators/common');
+
+
+describe('searchPermissions', () => {
+
+  describe('query', () => {
+    const query = schema.searchPermissions.query.describe();
+
+    it('requires at least 1 parameter', () => {
+      expect(query.rules).toEqual(expect.arrayContaining([
+        expect.objectContaining({
+          name: 'min',
+          args: {
+            limit: 1
+          }
+        })
+      ]));
+    });
+
+    describe('userId', () => {
+      const userId = query.keys.userId;
+
+      it('is the expected schema', () => {
+        expect(userId).toEqual(uuidv4MultiModel.describe());
+      });
+    });
+
+    describe('objId', () => {
+      const objId = query.keys.objId;
+
+      it('is the expected schema', () => {
+        expect(objId).toEqual(uuidv4MultiModel.describe());
+      });
+    });
+
+    describe('permCode', () => {
+      const permCode = query.keys.permCode;
+
+      it('is the expected schema', () => {
+        expect(permCode).toEqual(stringMultiModel.describe());
+      });
+    });
+  });
+});
+
+describe('listPermissions', () => {
+
+  describe('query', () => {
+    const query = schema.listPermissions.query.describe();
+
+    describe('userId', () => {
+      const userId = query.keys.userId;
+
+      it('is the expected schema', () => {
+        expect(userId).toEqual(uuidv4MultiModel.describe());
+      });
+    });
+
+    describe('objId', () => {
+      const objId = query.keys.objId;
+
+      it('is the expected schema', () => {
+        expect(objId).toEqual(uuidv4MultiModel.describe());
+      });
+    });
+
+    describe('permCode', () => {
+      const permCode = query.keys.permCode;
+
+      it('is the expected schema', () => {
+        expect(permCode).toEqual(stringMultiModel.describe());
+      });
+    });
```

Comment:
I can't quite remember, but permCode may be one of the string values we explicitly define as an enum on the API spec. If that is the case, we could go one step further and do string matches to ensure that only values that are within that enum set pass.